test/integrationtests/voight_kampff/tools.py

```diff
@@ -51,6 +51,35 @@ def then_wait(msg_type, criteria_func, context, timeout=TIMEOUT):
     return False, debug
 
 
+def then_wait_fail(msg_type, criteria_func, context, timeout=TIMEOUT):
+    """Wait for a specified time failing if criteria is fulfilled.
+
+    Arguments:
+        msg_type: message type to watch
+        criteria_func: Function to determine if a message fulfilling the
+                       test case has been found.
+        context: behave context
+        timeout: Time allowance for a message fulfilling the criteria
+
+    Returns:
+        tuple (bool, str) test status and debug output
+    """
+    start_time = time.monotonic()
+    debug = ''
+    test_failed = False
+    while time.monotonic() < start_time + timeout:
+        for message in context.bus.get_messages(msg_type):
+            status, test_dbg = criteria_func(message)
+            if status:
+                debug += test_dbg
+                context.matched_message = message
+                context.bus.remove_message(message)
+                return False, debug
+        context.bus.new_message_available.wait(0.5)
+    # Timed out return debug from test
+    return True, debug
```

Comment:
The `test_failed` variable doesn't sem to be used.

The answer to this is probably "No" but I'll ask anyway,

could this function be

```python
    def then_wait_fail(msg_type, criteria_func, context, timeout=TIMEOUT):
    """Wait for a specified time failing if criteria is fulfilled.
    [...]
    """
    res = then_wait(msg_type, criteria_func, context, timeout)
    return (not res[0], res[1])
```